pkg/openstack/loadbalancer_test.go

```diff
@@ -133,3 +134,299 @@ func TestPopListener(t *testing.T) {
 		assert.Equal(t, ids, item.result, item.name)
 	}
 }
+
+type testGetRulesToCreateAndDelete struct {
+	wantedRules   []rules.CreateOpts
+	existingRules []rules.SecGroupRule
+	toCreate      []rules.CreateOpts
+	toDelete      []rules.SecGroupRule
+}
+
+func TestGetRulesToCreateAndDelete(t *testing.T) {
+	items := []testGetRulesToCreateAndDelete{
+		// Empty elements
```

Comment:
nit: It's a common idiom to have the test description as part of the test. As an example, take a look at this randomly selected test in CAPO: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/9d183bd479fe9aed4f6e7ac3d5eee46681c518e7/pkg/cloud/services/compute/instance_test.go#L120-L121

Notice how the test name is part of the test data. Then when we run it we do:

```golang
	for _, tt := range tests {
		t.Run(tt.testName, func(t *testing.T) {
                  ...test function...
                }
        }
```

That way each invocation is a separate test with its own name, which is easier to read when it fails.

Reply:
Done!